File: app/src/main/java/com/maximatech/provaandroid/core/domain/model/Order.java
package com.maximatech.provaandroid.core.domain.model;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

public class Order {
    private int numeroPedRca;
    private String numeroPedErp;
    private String codigoCliente;
    private String nomeCliente;
    private String data;
    private String status;
    private String critica;
    private String tipo;
    private List<String> legendas;

    public static final Order EMPTY = new Order(
            0, "", "", "", "", "", "", "", new ArrayList<>()
    );

    public Order() {
        this.legendas = new ArrayList<>();
    }

    public Order(int numeroPedRca, String numeroPedErp, String codigoCliente,
                 String nomeCliente, String data, String status, String critica,
                 String tipo, List<String> legendas) {
        this.numeroPedRca = numeroPedRca;
        this.numeroPedErp = numeroPedErp;
        this.codigoCliente = codigoCliente;
        this.nomeCliente = nomeCliente;
        this.data = data;
        this.status = status;
        this.critica = critica;
        this.tipo = tipo;
        this.legendas = legendas != null ? legendas : new ArrayList<>();
    }

    public int getNumeroPedRca() { return numeroPedRca; }
    public String getNumeroPedErp() { return numeroPedErp; }
    public String getCodigoCliente() { return codigoCliente; }
    public String getNomeCliente() { return nomeCliente; }
    public String getData() { return data; }
    public String getStatus() { return status; }
    public String getCritica() { return critica; }
    public String getTipo() { return tipo; }
    public List<String> getLegendas() { return legendas; }

    public void setNumeroPedRca(int numeroPedRca) { this.numeroPedRca = numeroPedRca; }
    public void setNumeroPedErp(String numeroPedErp) { this.numeroPedErp = numeroPedErp; }
    public void setCodigoCliente(String codigoCliente) { this.codigoCliente = codigoCliente; }
    public void setNomeCliente(String nomeCliente) { this.nomeCliente = nomeCliente; }
    public void setData(String data) { this.data = data; }
    public void setStatus(String status) { this.status = status; }
    public void setCritica(String critica) { this.critica = critica; }
    public void setTipo(String tipo) { this.tipo = tipo; }
    public void setLegendas(List<String> legendas) {
        this.legendas = legendas != null ? legendas : new ArrayList<>();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Order order = (Order) o;
        return numeroPedRca == order.numeroPedRca &&
                Objects.equals(numeroPedErp, order.numeroPedErp) &&
                Objects.equals(codigoCliente, order.codigoCliente) &&
                Objects.equals(nomeCliente, order.nomeCliente) &&
                Objects.equals(data, order.data) &&
                Objects.equals(status, order.status) &&
                Objects.equals(critica, order.critica) &&
                Objects.equals(tipo, order.tipo) &&
                Objects.equals(legendas, order.legendas);
    }

    @Override
    public int hashCode() {
        return Objects.hash(numeroPedRca, numeroPedErp, codigoCliente, nomeCliente,
                data, status, critica, tipo, legendas);
    }

    @Override
    public String toString() {
        return "Order{" +
                "numeroPedRca=" + numeroPedRca +
                ", numeroPedErp='" + numeroPedErp + '\'' +
                ", codigoCliente='" + codigoCliente + '\'' +
                ", nomeCliente='" + nomeCliente + '\'' +
                ", data='" + data + '\'' +
                ", status='" + status + '\'' +
                ", critica='" + critica + '\'' +
                ", tipo='" + tipo + '\'' +
                ", legendas=" + legendas +
                '}';
    }
}

File: app/src/main/java/com/maximatech/provaandroid/core/domain/model/Order.kt
package com.maximatech.provaandroid.core.domain.model

data class Order(
    val numeroPedRca: Int,
    val numeroPedErp: String,
    val codigoCliente: String,
    val nomeCliente: String,
    val data: String,
    val status: String,
    val critica: String,
    val tipo: String,
    val legendas: List<String>
) {
    companion object {
        val EMPTY = Order(
            numeroPedRca = 0,
            numeroPedErp = "",
            codigoCliente = "",
            nomeCliente = "",
            data = "",
            status = "",
            critica = "",
            tipo = "",
            legendas = emptyList()
        )
    }
}

enum class OrderStatus {
    PROCESSADO,
    EM_PROCESSAMENTO,
    PENDENTE,
    CANCELADO;

    companion object {
        fun fromString(status: String): OrderStatus {
            return when (status.uppercase()) {
                "PROCESSADO" -> PROCESSADO
                "EM PROCESSAMENTO" -> EM_PROCESSAMENTO
                "PENDENTE" -> PENDENTE
                "CANCELADO" -> CANCELADO
                else -> PENDENTE
            }
        }
    }
}

enum class OrderType {
    PEDIDO,
    ORCAMENTO;

    companion object {
        fun fromString(type: String): OrderType {
            return when (type.uppercase()) {
                "PEDIDO" -> PEDIDO
                "ORCAMENTO" -> ORCAMENTO
                else -> PEDIDO
            }
        }
    }
}

enum class OrderLegend(val displayName: String) {
    PEDIDO_SOFREU_CORTE("Pedido sofreu corte"),
    PEDIDO_FEITO_TELEMARKETING("Pedido feito pelo Telemarketing"),
    PEDIDO_CANCELADO_ERP("Pedido cancelado no ERP"),
    PEDIDO_COM_FALTA("Pedido com falta"),
    PEDIDO_COM_DEVOLUCAO("Pedido com devolução"),
    EM_PROCESSAMENTO_FV("Em processamento por parte do FV"),
    PEDIDO_RECUSADO_ERP("Pedido recusado pelo ERP"),
    POSICAO_ERP_PENDENTE("Posição no ERP Pendente"),
    POSICAO_ERP_BLOQUEADO("Posição no ERP Bloqueado"),
    POSICAO_ERP_LIBERADO("Posição no ERP Liberado"),
    POSICAO_ERP_MONTADO("Posição no ERP Montado"),
    POSICAO_ERP_FATURADO("Posição no ERP Faturado"),
    POSICAO_ERP_CANCELADO("Posição no ERP Cancelado");

    companion object {
        fun fromString(legend: String): OrderLegend? {
            return values().find { it.name == legend }
        }
    }
}